### PR TITLE
Reduced the number of calls to generic_getLinks.

### PR DESCRIPTION
**Pull request by MiguelRocha**
Refactored generic_getLinks to parse all src and href type links.
Removed getDependencies()
Make zimcheck call generic_getLinks only once

### CODE REVIEW

**Comment by kelson42** [comment]
@MiguelRocha Do we have a ticket for that? Is that a speed-up PR?

File: src/zimcheck.cpp [hunk 1]
```diff
@@ -361,13 +356,14 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
             if (it->getMimeType() != "text/html")
                 continue;
 
-            auto links = getDependencies(it->getPage());
-            for (auto &link: links)
+            if (url_check == false)
+                links = generic_getLinks(it->getPage());
```
Comment (mgautierfr) [inline src/zimcheck.cpp:360]: Maybe move this in line 289 with :
```
if (url_check || url_check_external) {
    links = generic_getLinks(it->getData())
}
```
(`getPage` will be removed in next api)

File: src/tools.h [hunk 2]
```diff
@@ -25,6 +25,12 @@
 #include <string>
 #include <vector>
 
+typedef struct html_link
+{
+    std::string attribute;
+    std::string link;
+} html_link;
+
```
Comment (mgautierfr) [inline src/tools.h:33]: I have to say that I would have naively use a `std::tuple` or a `std::pair`, but it is better with a specific structure.

**Review by mgautierfr — CHANGES_REQUESTED** [review]
Just a small change about where we initialize `links`.
But else we are ok.

File: src/zimcheck.cpp (hunk 1 above)
Comment (MiguelRocha) [reply src/zimcheck.cpp]: done

File: src/tools.h (hunk 2 above)
Comment (MiguelRocha) [reply src/tools.h:33]: I was thinking about that as well but I wanted to be more expressive here to avoid confusion.

**Comment by MiguelRocha** [comment]
> @MiguelRocha Do we have a ticket for that? Is that a speed-up PR?

Yes it is the first point of https://github.com/openzim/zim-tools/issues/134

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**